src/lib/pubkey/classic_mceliece/cmce_parameters.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Classic McEliece Parameters
+ * (C) 2023 Jack Lloyd
+ *     2023,2024 Fabian Albert, Amos Treiber - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ **/
+
+#include <botan/internal/cmce_parameters.h>
+#include <botan/internal/cmce_poly.h>
+
+namespace Botan {
+
+namespace {
+
+std::vector<Classic_McEliece_Polynomial_Ring::Big_F_Coefficient> determine_big_f_coef(size_t t, GF_Mod modulus) {
+   std::vector<Classic_McEliece_Polynomial_Ring::Big_F_Coefficient> big_f_coef;
+   switch(t) {
+      // TODO: Remove test instance on final PR
+      case 8:  //y^8 + y^4 + y^3 + y^2 + 1 (test instances)
+         big_f_coef.push_back({0, Classic_McEliece_GF(GF_Elem(1), modulus)});
+         big_f_coef.push_back({2, Classic_McEliece_GF(GF_Elem(1), modulus)});
+         big_f_coef.push_back({3, Classic_McEliece_GF(GF_Elem(1), modulus)});
+         big_f_coef.push_back({4, Classic_McEliece_GF(GF_Elem(1), modulus)});
+         break;
```

Comment:
Instead of multiple `push_back` into a pre-defined vector (that might cause multiple re-allocs), you could just implicitly call the constructor of the returned vector in-place:

```suggestion

         return {{0, Classic_McEliece_GF(GF_Elem(1), modulus)},
                 {2, Classic_McEliece_GF(GF_Elem(1), modulus)},
                 {3, Classic_McEliece_GF(GF_Elem(1), modulus)},
                 {4, Classic_McEliece_GF(GF_Elem(1), modulus)}};
```

... goes without saying that this would need to be adapted for the other cases.